instrumentation/opentelemetry-instrumentation-grpc/src/opentelemetry/instrumentation/grpc/_server.py

```diff
@@ -125,18 +126,16 @@ def set_code(self, code):
         self.code = code
         # use details if we already have it, otherwise the status description
         details = self.details or code.value[1]
+        self._active_span.set_attribute("rpc.grpc.status_code", code.name)
         self._active_span.set_status(
-            Status(status_code=StatusCode(code.value[0]), description=details)
+            Status(status_code=StatusCode.ERROR, description=details)
         )
         return self._servicer_context.set_code(code)
 
     def set_details(self, details):
         self.details = details
         self._active_span.set_status(
-            Status(
-                status_code=StatusCode(self.code.value[0]),
-                description=details,
-            )
+            Status(status_code=StatusCode.ERROR, description=details)
```

Comment:
I can see that everywhere we use `set_details` it's to add an error message so this change makes sense.

Although I feel like a good change (may or may not be for this PR) would be to have the method renamed to `set_error_details`?

Reply:
This class is a subclass of `grpc.ServicerContext` and so we should probably stay compliant with that API.

Reply:
It's subclassing servicer context so can't be changed https://grpc.github.io/grpc/python/grpc.html#grpc.ServicerContext.set_details